### PR TITLE
#3780: Do not display wrong warning when loading ".js" or ".json"

### PR DESCRIPTION
**Pull request by BR0kEN-**
Issue: https://github.com/storybooks/storybook/issues/3780
Description: https://github.com/storybooks/storybook/issues/3780#issuecomment-402491084

### CODE REVIEW

File: lib/core/src/server/serverRequire.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ export default function serverRequire(filePath) {
 
   const moduleDescriptor = interpret.extensions[candidateExt];
 
-  if (registerCompiler(moduleDescriptor) === 0) {
+  if (moduleDescriptor !== null && registerCompiler(moduleDescriptor) === 0) {
```
Comment (igor-dv) [inline lib/core/src/server/serverRequire.js]: So, is the problem only with a `.json`? because there is already a check ☝️  for `.js`.

Comment (BR0kEN-) [reply lib/core/src/server/serverRequire.js]: Yeah, you are right. My proposition to not show the warning when `moduleDescriptor` is `undefined` (no key in `interpret.extensions`) or `null` (as for `.js` and `.json`).

**Review by igor-dv — APPROVED** [review]
Thanks. we are having some troubles with CI, once it fixed I will merge.

**Comment by igor-dv** [comment]
Can you please merge from master?

**Comment by BR0kEN-** [comment]
Done. Let's wait for the builds.

**Pull request merged**